packages/perseus/src/widgets/interactive-graphs/mafs-graph.test.tsx

```diff
@@ -0,0 +1,93 @@
+import {screen, render, waitFor} from "@testing-library/react";
+import {userEvent as userEventLib} from "@testing-library/user-event";
+import React from "react";
+
+import {ApiOptions} from "../../perseus-api";
+
+import {MafsGraph} from "./mafs-graph";
+
+import type {InteractiveGraphProps} from "./types";
+import type {PerseusInteractiveGraphWidgetOptions} from "../../perseus-types";
+import type {UserEvent} from "@testing-library/user-event";
+
+function getBaseMafsGraphProps(): InteractiveGraphProps {
+    const mafsGraphBaseProps: PerseusInteractiveGraphWidgetOptions = {
+        step: [1, 1],
+        gridStep: [1, 1],
+        snapStep: [1, 1],
+        markings: "graph",
+        labels: [],
+        showProtractor: false,
+        showRuler: false,
+        rulerLabel: "",
+        range: [
+            [-10, 10],
+            [-10, 10],
+        ],
+        graph: {type: "segment"},
+        correct: {type: "segment"},
+        rulerTicks: 4,
+    };
+
+    const widgetBaseProps: InteractiveGraphProps = {
+        ...mafsGraphBaseProps,
+        alignment: null,
+        apiOptions: {
+            ...ApiOptions.defaults,
+        },
+        containerSizeClass: "small",
+        findWidgets: () => [],
+        isLastUsedWidget: false,
+        onBlur: () => {},
+        onChange: () => {},
+        onFocus: () => {},
+        problemNum: 1,
+        reviewModeRubric: {
+            ...mafsGraphBaseProps,
+        },
+        static: true,
+        trackInteraction: () => {},
+        widgetId: "mafs-graph",
+        linterContext: {} as any,
+    };
+
+    return widgetBaseProps;
```

Comment:
I had no idea `MafsGraph` required all these props! I think very few of them are actually used.

I think I would prefer it if, instead of having to create these do-nothing props in tests, we trimmed down the props type of MafsGraph to include only what it needs. Currently, the props type is derived from the InteractiveGraph props:

```ts
React.PropsWithChildren<InteractiveGraphProps> & {box: [number, number]}
```

Reply:
I'm a little confused with this one. `widgetBaseProps` is based on `WidgetProps` which is used by all widgets. `mafsGraphBaseProps` is based on `PerseusInteractiveGraphWidgetOptions` which is used by the legacy Interaction widget. I'd be concerned about modifying either of those due to impact on non-Mafs widgets.

I could nest `MafsGraph` - the parent could take `PerseusInteractiveGraphWidgetOptions` and pick out the pieces that the child `MafsGraph` needs. That way the parent could be easily swapped with the original Interactive Graph widget and the child could focus only on what it requires to function?

I'm tempted to suggest that be a follow-up though.

Reply:
I think we can change the types and tests without having to change any production code. Here's a commit showing what I mean. You should be able to cherry-pick it into your branch, if you agree it's an improvement: https://github.com/Khan/perseus/commit/cfee1eaa3e4a9812a92696d0cff2e52220fa5dcc